components/ShoppingItem/ShoppingItem.js

```diff
@@ -3,49 +3,63 @@ import Link from "next/link";
 import { useState } from "react";
 
 export default function ShoppingItem({
-  id,
   onDeleteItem,
   shoppingItem,
   onEditItem,
   onChangeMode,
-  mode,
+  onToggleIsPurchased,
+  purchasedItems,
 }) {
   const [isToBeDeleted, setIsToBeDeleted] = useState(false);
 
-  // const [isToBeEdited, setIsToBeEdited] = useState(false);
-
   function toggleIsToBeDeleted() {
     setIsToBeDeleted(!isToBeDeleted);
   }
 
-  // function toggleIsToBeEdited() {
-  //   setIsToBeEdited(!isToBeEdited);
-  // }
-
   return (
     <ListItem
-      key={shoppingItem.id}
       $backgroundColor={shoppingItem.backgroundColor}
+      $isPurchased={shoppingItem.isPurchased}
     >
       <CategoryBox>{shoppingItem.category}</CategoryBox>
-
       {!isToBeDeleted ? (
         <>
-          {shoppingItem.name}: {shoppingItem.quantity}
+          <StyledTickboxNameNumberContainer>
+            <form>
+              <input
+                type="checkbox"
+                id={`checkbox-${shoppingItem.id}`}
+                checked={shoppingItem.isPurchased}
+                onChange={() => {
+                  onToggleIsPurchased(shoppingItem.id);
+                }}
+              />
+              <label htmlFor={`checkbox-${shoppingItem.id}`}>
+                {purchasedItems.includes(shoppingItem.id) ? "purchased" : ""}
```

Comment:
You are checking here, if the shopping item is purchased. That information is saved inside your `shoppingItem`, as each `shoppingItem` has an `isPurchased` key. So there is no need to pass down the `purchasedItems` for this check.

```suggestion
                {shoppingItem.isPurchased ? "purchased" : ""}
```

Reply:
Done earlier